code/modules/shuttle/emergency.dm

```diff
@@ -161,6 +166,21 @@
 /obj/machinery/computer/emergency_shuttle/proc/clear_recent_action(mob/user)
 	acted_recently -= user
 
+/obj/machinery/computer/emergency_shuttle/proc/pick_emag_amount()
+	switch(rand(1, 5))
```

Comment:
The numbers below should be defines and undefines for readability.
Instead of "2", it can be #define DOUBLE_TIME 2
Things along those lines.